server/routes/incentiveLevels.ts

```diff
@@ -1,32 +1,62 @@
-import type { RequestHandler, Router } from 'express'
+import type { RequestHandler, Response, Router } from 'express'
+import jwtDecode from 'jwt-decode'
 
+import logger from '../../logger'
 import asyncMiddleware from '../middleware/asyncMiddleware'
+import authorisationMiddleware from '../middleware/authorisationMiddleware'
 import { IncentivesApi } from '../data/incentivesApi'
 
+const role = 'ROLE_MAINTAIN_INCENTIVE_LEVELS'
+const requireRole = authorisationMiddleware([role])
+const hasRole = (res: Response) => {
+  const { authorities: roles = [] } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
+  return roles && roles.includes(role)
+}
```

Comment:
I don't know if this is changing a lot in the subsequent commits but I have a couple of small observations:
- `role` is quite generic for a constant
- `role` seems to only be used inside `hasRole`
- you could make that constant more explicit, e.g. call it `ROLE_MAINTAIN_INCENTIVE_LEVELS` (I know that would be long)
- ...or move it inside `hasRole()`
- `hasRole()` in this context literally mean `canManageIncentiveLevels()`
- however I see there is another `hasRole()` function which is doing pretty much the same thing. Maybe this could be extracted in a separate module responsible for role checking?
- _Maybe_ these roles constants could be moved in this module in an enum or something
- I also see a similar logic is in the `authorisationMiddleware`. This comes from the TS template IIRC so I don't know if you want to touch it much but it could also reuse this `hasRole()` function

Reply:
nah that's fair. i just wrote it generically so the code could be copied exactly into the other route, with only the constant value being different.

initially, i had wanted so higher-up middleware to set the roles on the user object (somewhere) but thought that would be changing too much.

will look at better naming